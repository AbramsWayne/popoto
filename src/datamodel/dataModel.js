var dataModel = {};

dataModel.nodes = [];
dataModel.links = [];

export default dataModel