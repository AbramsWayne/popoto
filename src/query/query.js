var query = {};
export default query