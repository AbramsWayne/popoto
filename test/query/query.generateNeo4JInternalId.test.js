import provider from "../../src/provider/provider";
import query from '../../src/query/query.js'
import dataModel from "../../src/datamodel/dataModel";

describe("one branch", function () {
    beforeEach(() => {
        dataModel.nodes = [
            {
                id: 0,
                label: "Root",
                internalLabel: "root",
                type: 0
            },
            {
                id: 1,
                label: "Node",
                internalLabel: "node1",
                type: 1
            },
            {
                id: 2,
                label: "Node",
                internalLabel: "node2",
                value: [{label: "Node", internalID: 102, attributes: {Nname: "name2"}}],
                type: 1,
            },
            {
                id: 3,
                label: "Node",
                internalLabel: "node3",
                type: 1,
                value: [
                    {label: "Node", internalID: 1031, attributes: {Nname: "name31"}},
                    {label: "Node", internalID: 1032, attributes: {Nname: "name32"}}
                ],
            }
        ];

        dataModel.links = [
            {
                id: 0,
                label: "LINKED_TO1",
                source: dataModel.nodes[0],
                target: dataModel.nodes[1],
                type: 1
            },
            {
                id: 1,
                label: "LINKED_TO2",
                source: dataModel.nodes[1],
                target: dataModel.nodes[2],
                type: 1
            },
            {
                id: 0,
                label: "LINKED_TO3",
                source: dataModel.nodes[0],
                target: dataModel.nodes[3],
                type: 1
            },
        ];

        provider.node.Provider = {
            "Root": {
                returnAttributes: [query.NEO4J_INTERNAL_ID],
                constraintAttribute: query.NEO4J_INTERNAL_ID,
            },
            "Node": {
                returnAttributes: ["Nname", query.NEO4J_INTERNAL_ID],
                constraintAttribute: query.NEO4J_INTERNAL_ID,
            }
        };
    });

    afterEach(() => {
        delete provider.node.Provider;
    });

    test("generateResultQuery", () => {
        var generateResultQuery = query.generateResultQuery(false);
        expect(generateResultQuery.statement).toMatchSnapshot();
        expect(generateResultQuery.parameters).toMatchSnapshot();
    });
});
