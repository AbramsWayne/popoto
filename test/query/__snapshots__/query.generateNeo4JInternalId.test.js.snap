// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`one branch generateResultQuery 1`] = `"MATCH (root:\`Root\`), (node1:\`Node\`)-[r0:\`LINKED_TO2\`]->(node2:\`Node\`), (root:\`Root\`)-[r1:\`LINKED_TO3\`]->(node3:\`Node\`), (root:\`Root\`)-[r2:\`LINKED_TO1\`]->(node1:\`Node\`) WHERE ID(node2) = $node2_internalID AND ID(node3) IN $node3_internalID RETURN DISTINCT ID(root) AS NEO4JID LIMIT $limit"`;

exports[`one branch generateResultQuery 2`] = `
Object {
  "limit": 100,
  "node2_internalID": 102,
  "node3_internalID": Array [
    1031,
    1032,
  ],
}
`;
